lib/launcher.ts

```diff
@@ -243,7 +241,13 @@ let initFn = function(configFile: string, additionalConfig: Config) {
           }
         }
 
-        let deferred = q.defer<any>();  // Resolved when all tasks are completed
+        let deferredResolve: () => void;
```

Comment:
There is also a consideration of whether an exception is to be swallowed or thrown out of the method.